File: src/net/minecraft/client/render/entity/SpiderEntityRenderer.cpp
#include "net/minecraft/client/gl/GlConstants.hpp"
#include "net/minecraft/client/render/RenderCore.hpp"
#include "net/minecraft/client/render/entity/EntityRenderers.hpp"
#include "net/minecraft/client/render/entity/model/SpiderEntityModel.hpp"
#include "net/minecraft/entity/LivingEntity.hpp"
#include "net/minecraft/entity/mob/SpiderEntity.hpp"
namespace net::minecraft::client::render::entity {
SpiderEntityRenderer::SpiderEntityRenderer() : LivingEntityRenderer(new model::SpiderEntityModel(), 1.0f) {
 setDecorationModel(new model::SpiderEntityModel());
}
float SpiderEntityRenderer::getDeathYaw(const net::minecraft::LivingEntity& entity) const {
 (void)entity;
 return 180.0f;
}
bool SpiderEntityRenderer::bindTexture(const net::minecraft::LivingEntity& entity, int layer, float tickDelta) {
 if(layer != 0) {
  return false;
 }
 EntityRenderer::bindTexture("/mob/spider_eyes.png");
 // MCP RenderSpider: getEntityBrightness(1.0F) — hardcoded 1.0F partial tick.
 const float brightness = entity.getBrightnessAtEyes(1.0f);
 const float alpha = (1.0f - brightness) * 0.5f;
 core::blendAlpha();
 render::core::setAlphaTestRef(0.0f);
 render::core::setConstColor(1.0f, 1.0f, 1.0f, alpha);
 (void)tickDelta;
 return true;
}
} // namespace net::minecraft::client::render::entity
#include "net/minecraft/client/entity/EntityClientRendererRegistration.hpp"
#include "net/minecraft/entity/mob/SpiderEntity.hpp"
namespace net::minecraft::entity::mob {
std::unique_ptr<::net::minecraft::client::render::entity::EntityRenderer> SpiderEntity::ClientRenderer::create() {
 return std::make_unique<::net::minecraft::client::render::entity::SpiderEntityRenderer>();
}
} // namespace net::minecraft::entity::mob
namespace {
static ::net::minecraft::registry::RegisterEntityRenderer<net::minecraft::entity::mob::SpiderEntity> autoRendererReg;
} // namespace

File: mcp/src/net/minecraft/client/render/entity/SpiderEntityRenderer.java
/*
 * Decompiled with CFR 0.1.1 (FabricMC 57d88659).
 * 
 * Could not load the following classes:
 *  net.fabricmc.api.EnvType
 *  net.fabricmc.api.Environment
 */
package net.minecraft.client.render.entity;

import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.minecraft.client.render.entity.LivingEntityRenderer;
import net.minecraft.client.render.entity.model.SpiderEntityModel;
import net.minecraft.entity.mob.SpiderEntity;
import org.lwjgl.opengl.GL11;

@Environment(value=EnvType.CLIENT)
public class SpiderEntityRenderer
extends LivingEntityRenderer {
    public SpiderEntityRenderer() {
        super(new SpiderEntityModel(), 1.0f);
        this.setDecorationModel(new SpiderEntityModel());
    }

    protected float getDeathYaw(SpiderEntity arg) {
        return 180.0f;
    }

    protected boolean bindTexture(SpiderEntity arg, int i, float f) {
        if (i != 0) {
            return false;
        }
        if (i != 0) {
            return false;
        }
        this.bindTexture("/mob/spider_eyes.png");
        float f2 = (1.0f - arg.getBrightnessAtEyes(1.0f)) * 0.5f;
        GL11.glEnable(3042);
        GL11.glDisable(3008);
        GL11.glBlendFunc(770, 771);
        GL11.glColor4f(1.0f, 1.0f, 1.0f, f2);
        return true;
    }
}

